### PR TITLE
Throw exception when set cookie in 'about:blank' page

### PR DESCRIPTION
**Pull request by jihchi**
Original from https://github.com/GoogleChrome/puppeteer/pull/1411#issuecomment-345423707

When you set cookie in 'about:blank' page, you'll get following exception:

```
AssertionError [ERR_ASSERTION]: Blank page can not have cookie "<cookie_name>"
```

### CODE REVIEW

File: lib/Page.js [hunk 1]
```diff
@@ -255,11 +255,13 @@ class Page extends EventEmitter {
...
+    const pageURL = this.url();
+    const startsWithHTTP = pageURL.startsWith('http');
     const items = cookies.map(cookie => {
       const item = Object.assign({}, cookie);
-      const pageURL = this.url();
-      if (!item.url && pageURL.startsWith('http'))
-        item.url = this.url();
+      if (!item.url && startsWithHTTP)
+        item.url = pageURL;
+      console.assert(item.url !== 'about:blank', `Blank page can not have cookie "${item.name}"`);
```
Comment (JoelEinbinder) [inline lib/Page.js]: I think data urls also cannot have cookies.

**Review by JoelEinbinder — APPROVED** [review]
Looks good! Thanks for the pull request.

File: lib/Page.js (hunk 1 above)
Comment (jihchi) [reply lib/Page.js]: Fixed in commit f1f83c448f73a7ae1e032b838e250d82cd514055

**Comment by jihchi** [comment]
Hi @JoelEinbinder , @aslushnikov ,

Anything else need to fix or change ?

**Comment by aslushnikov** [comment]
@jihchi unfortunately, the "resolve conflicts" button didn't help me resolve conflicts in this branch (not sure why). Can you please rebaseline the branch atop of `master`?

**Comment by jihchi** [comment]
@aslushnikov done.

**Pull request merged**